File: LeetCode/169.majority-element.py
#
# @lc app=leetcode id=169 lang=python3
#
# [169] Majority Element
#
from typing import List


from collections import Counter
# @lc code=start
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = Counter(nums)
        l = list(count)
        l.sort(key=lambda x:x[1], reverse=True)
        return l[0][0]
# @lc code=end

File: LeetCode/169.majority-element.java
/*
 * @lc app=leetcode id=169 lang=java
 *
 * [169] Majority Element
 */
import java.util.ArrayList;
import java.util.HashMap;
// @lc code=start
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> l = new HashMap<>();
        int max_ = nums[0];
        int times = 0;
        for (int i : nums) {
            if (l.get(i) == null) {
                l.put(i, 1);
                continue;
            };
            l.replace(i, l.get(i) + 1);
            if (l.get(i) > times)  {
                max_ = i;
                times = l.get(i);
            }
        }

        return max_;

        
    }
}
// @lc code=end

